media-video/wireplumber/wireplumber-9999.ebuild

```
@@ -0,0 +1,80 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+LUA_COMPAT=( lua5-{3,4} )
+
+inherit lua-single meson
+
+if [[ ${PV} == 9999 ]]; then
+	EGIT_REPO_URI="https://gitlab.freedesktop.org/pipewire/${PN}.git"
+	EGIT_BRANCH="master"
+	inherit git-r3
+else
+	SRC_URI="https://gitlab.freedesktop.org/pipewire/${PN}/-/archive/${PV}/${P}.tar.gz"
+	KEYWORDS="~amd64"
+fi
+
+DESCRIPTION="Replacement for pipewire-media-session"
+HOMEPAGE="https://gitlab.freedesktop.org/pipewire/wireplumber"
+
+LICENSE="MIT"
+SLOT="0/0.4"
+IUSE="systemd"
+
+REQUIRED_USE="${LUA_REQUIRED_USE}"
+
+# introspection? ( dev-libs/gobject-introspection ) is valid but likely only used for doc building
+BDEPEND="
+	dev-libs/glib
+	dev-util/gdbus-codegen
+	dev-util/glib-utils
+"
+
+DEPEND="
+	${LUA_DEPS}
+	>=dev-libs/glib-2.62
+	>=media-video/pipewire-0.3.26
+	virtual/libc
+	systemd? ( sys-apps/systemd )
+
+"
+
+# Any dev-lua/* deps get declared like this inside RDEPEND:
+#	$(lua_gen_cond_dep '
+#		dev-lua/<NAME>[${LUA_USEDEP}]
+#	')
+RDEPEND="${DEPEND}"
+
+DOCS=( {NEWS,README}.rst )
+
+src_configure() {
+	local emesonargs=(
+		-Dintrospection=disabled # Likely only used for doc generation
+		-Dsystem-lua=true # We always unbundle everything we can
+		#-Dsystem-lua-version=<version number> # Some distros can't enforce deterministc system Lua themselves
```

Should we do:
$(lua_get_version)
```suggestion
		-Dsystem-lua-version=$(lua_get_version)
```
instead, just for future proofing? Or no need because it respects some environment variable the eclass sets?

---

If the eclass couldn't enforce it, then the SINGLE_LUA_TARGET would be a sham, so I just hand-waved it away by assuming that the  lua-single.eclass is better than certain other distros and can enforce the chosen version. :sweat_smile:

As you might be aware, Lua has no concept of consistent version naming, causing each distro to came up with their own pkg-config file naming scheme. Gentoo probably didn't want to be left out and for 5.4 implemented all(?) of them: `lua-5.4.pc lua5.4.pc lua54.pc`one of which we need pass to that Meson switch (minus the file extension).

For now I'll install 5.3 alongside 5.4 and do some configure runs to see if it reliably picks the version requested (I may or may not have done this at some point) in hopes of not having to bash `lua_get_version()` into one of the accepts formats.